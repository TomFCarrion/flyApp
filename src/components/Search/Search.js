import React from 'react';
import DayPickerInput from 'react-day-picker/DayPickerInput';
import { useDispatch, useSelector } from 'react-redux';
import axios from 'axios';
import { Link } from 'react-router-dom';
import moment from 'moment';

import {
  SET_FROM,
  SET_TO,
  SET_START_DATE,
  SET_END_DATE,
} from '../../actions/actionTypes';
import { formatDate, parseDate } from 'react-day-picker/moment';

import 'react-day-picker/lib/style.css';
require('./styles/search.scss');

const Search = () => {
  const from = useSelector((state) => state.infoReducer.from);
  const to = useSelector((state) => state.infoReducer.to);
  const startDate = useSelector((state) => state.infoReducer.startDate);
  const endDate = useSelector((state) => state.infoReducer.endDate);

  const dispatch = useDispatch();

  const handleInputChange = (value, action) => {
    dispatch({ type: action, data: value }), [dispatch];
  };

  const handleInputChange = (value, action) => {
    dispatch({ type: action, data: value }), [dispatch];
  };
  const handleCalendarChange = (value, action) => {
    dispatch({ type: action, data: moment(value).format('YYYY-MM-DD') }),
      [dispatch];
  };

  const modifiers = { start: from, end: to };

  return (
    <div className="wrapper">
      <input
        type="text"
        value={from}
        onChange={(e) => handleInputChange(e.target.value, SET_FROM)}
        name="from"
      />
      <input
        type="text"
        value={to}
        onChange={(e) => handleInputChange(e.target.value, SET_TO)}
        name="to"
      />
      <DayPickerInput
        value={startDate}
        placeholder="From"
        formatDate={formatDate}
        parseDate={parseDate}
        dayPickerProps={{
          modifiers,
          numberOfMonths: 2,
        }}
        onDayChange={(e) => handleCalendarChange(e, SET_START_DATE)}
      />{' '}
      â€”{' '}
      <span className="InputFromTo-to">
        <DayPickerInput
          value={endDate}
          placeholder="To"
          formatDate={formatDate}
          parseDate={parseDate}
          dayPickerProps={{
            modifiers,
            numberOfMonths: 2,
          }}
          onDayChange={(e) => handleCalendarChange(e, SET_END_DATE)}
        />
      </span>
      <Link to="/result">
        <button type="button">Click Me!</button>
      </Link>
    </div>
  );
};

export default Search;
