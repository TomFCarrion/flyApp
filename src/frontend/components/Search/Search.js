import React from 'react';
import DayPickerInput from 'react-day-picker/DayPickerInput';
import { useDispatch, useSelector } from 'react-redux';
import axios from 'axios';
import { Link } from 'react-router-dom';
import moment from 'moment';

import { formatDate, parseDate } from 'react-day-picker/moment';
import takeoff from '../../assets/takeOff.svg';
import landing from '../../assets/landing.svg';

import {
  SET_FROM,
  SET_FROM_ID,
  SET_TO,
  SET_TO_ID,
  SET_START_DATE,
  SET_END_DATE,
} from '../../actions/actionTypes';

import 'react-day-picker/lib/style.css';

require('./styles/search.scss');

const Search = () => {
  const from = useSelector(state => state.infoReducer.from);
  const to = useSelector(state => state.infoReducer.to);
  const startDate = useSelector(state => state.infoReducer.startDate);
  const endDate = useSelector(state => state.infoReducer.endDate);

  const dispatch = useDispatch();

  const getPlaceCode = (value, action) => {
    const options = {
      method: 'GET',
      url:
        'https://skyscanner-skyscanner-flight-search-v1.p.rapidapi.com/apiservices/autosuggest/v1.0/US/USD/en-US/',
      params: { query: value },
      headers: {
        'x-rapidapi-key': '08eb576b39mshaa3eb3f0249657ep1b15f7jsn598ba72db2e4',
        'x-rapidapi-host':
          'skyscanner-skyscanner-flight-search-v1.p.rapidapi.com',
      },
    };

    axios
      .request(options)
      .then((res) => {
        dispatch({ type: action, data: res.data.Places[0].PlaceId }),
        [dispatch];
      })
      .catch((error) => {
        console.error(error);
      });
  };

  const handleBlur = (e, action) => {
    if (!e.currentTarget.contains(e.relatedTarget)) {
      getPlaceCode(e.target.value, action);
    }
  };

  const handleInputChange = (value, action) => {
    dispatch({ type: action, data: value }), [dispatch];
  };
  const handleCalendarChange = (value, action) => {
    dispatch({ type: action, data: moment(value).format('YYYY-MM-DD') }),
    [dispatch];
  };

  const modifiers = { start: from, end: to };

  return (
    <div className='wrapper'>
      <div>
        <span className='icons'>
          <img className='icon' src={takeoff} width='100' height='50' />
          <img className='icon' src={landing} width='100' height='50' />
        </span>

        <span className='inputs'>

          <input
            placeholder='From'
            type='text'
            value={from}
            onChange={e => handleInputChange(e.target.value, SET_FROM)}
            onBlur={e => handleBlur(e, SET_FROM_ID)}
            name='from'
            data-cy='cy-from'
          />
          <input
            placeholder='To'
            type='text'
            value={to}
            onChange={e => handleInputChange(e.target.value, SET_TO)}
            onBlur={e => handleBlur(e, SET_TO_ID)}
            name='to'
            data-cy='cy-to'

          />
        </span>
      </div>

      <div className='Date'>
        <div className='InputFrom-from'>
          <DayPickerInput
            value={startDate}
            id='cy-depart'
            placeholder='Depart'
            formatDate={formatDate}
            parseDate={parseDate}
            dayPickerProps={{
              modifiers,
              numberOfMonths: 1,
              disabledDays: { before: new Date() },
            }}
            onDayChange={e => handleCalendarChange(e, SET_START_DATE)}
          />
        </div>
        <span className='InputFromTo-to'>
          <DayPickerInput
            value={endDate}
            placeholder='Return'
            data-cy='cy-return'
            formatDate={formatDate}
            parseDate={parseDate}
            dayPickerProps={{
              modifiers,
              numberOfMonths: 1,
              preventOverflow: {
                enabled: true,
              },
              disabledDays: { before: new Date() },
            }}
            popperModifiers={{
              preventOverflow: {
                enabled: true,
              },
            }}
            onDayChange={e => handleCalendarChange(e, SET_END_DATE)}

          />
        </span>
      </div>

      <Link to='/result'>
        <span className='button-Wrapper' data-cy='search-button'>
          {' '}
          <button className='button' type='button'>Get My Flight!</button>
        </span>
      </Link>
    </div>
  );
};

export default Search;
