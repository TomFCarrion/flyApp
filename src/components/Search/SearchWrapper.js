import React from 'react';
import Search from './Search';
require('./styles/searchWrapper.scss');

const SearchWrapper = () => {
  return (
    <div>
      <h1 className="title"> Search Your Flight</h1>
      <Search />
    </div>
  );
};
export default SearchWrapper;
