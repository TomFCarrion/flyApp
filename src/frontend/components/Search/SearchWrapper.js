import React from 'react';
import Search from './Search';
require('./styles/searchWrapper.scss');
import mainIllustration from '../../assets/mainIllustrations.svg'

const SearchWrapper = () => {
  return (
    <div className="searchWrapper">
      <div className="title">
        <img className="image" src={mainIllustration} />
      </div>
      <Search />
    </div>
  );
};
export default SearchWrapper;
