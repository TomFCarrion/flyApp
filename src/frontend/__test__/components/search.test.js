import React from 'react';
import { shallow } from 'enzyme';
import { create } from 'react-test-renderer';
import Search from '../../components/Search/Search';
import ProviderMock from '../../__mocks__/providerMock';

describe(' <Search/>', () => {
  test('SearchWrapper component render', () => {
    const search = shallow(
      <ProviderMock>
        <Search />
      </ProviderMock>,
    );
    expect(search.length).toEqual(1);
  });
});

describe('Search snapshot', () => {
  test('Check Search snapshot', () => {
    const search = create(
      <ProviderMock>
        <Search />
      </ProviderMock>,
    );
    expect(search.toJSON()).toMatchSnapshot();
  });
});
