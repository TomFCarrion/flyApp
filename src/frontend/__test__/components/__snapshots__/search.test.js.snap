// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search snapshot Check Search snapshot 1`] = `
<div
  className="wrapper"
>
  <div>
    <span
      className="icons"
    >
      <img
        className="icon"
        height="50"
        src={[Function]}
        width="100"
      />
      <img
        className="icon"
        height="50"
        src={[Function]}
        width="100"
      />
    </span>
    <span
      className="inputs"
    >
      <input
        data-cy="cy-from"
        name="from"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="From"
        type="text"
        value=""
      />
      <input
        data-cy="cy-to"
        name="to"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="To"
        type="text"
        value=""
      />
    </span>
  </div>
  <div
    className="Date"
  >
    <div
      className="InputFrom-from"
    >
      <div
        className="DayPickerInput"
      >
        <input
          onBlur={[Function]}
          onChange={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          placeholder="Depart"
          value=""
        />
      </div>
    </div>
    <span
      className="InputFromTo-to"
    >
      <div
        className="DayPickerInput"
      >
        <input
          onBlur={[Function]}
          onChange={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          placeholder="Return"
          value=""
        />
      </div>
    </span>
  </div>
  <a
    href="/result"
    onClick={[Function]}
  >
    <span
      className="button-Wrapper"
      data-cy="search-button"
    >
       
      <button
        className="button"
        type="button"
      >
        Get My Flight!
      </button>
    </span>
  </a>
</div>
`;
